docs/features/software-catalog/external-integrations.md

```diff
@@ -171,3 +171,97 @@ export default async function createPlugin(
 
 Start up the backend - it should now start reading from the previously
 registered location and you'll see your entities start to appear in Backstage.
+
+## Caching processing results
+
+The catalog periodically refresh entities in the catalog and by doing so it call
+out to external systems to fetch changes. This can be taxing for upstream
+services and large deployments may get rate limited if too many requests are
+sent. Luckily many external systems provide ETag support to check for changes
+which usually don't count towards the quota and saves resources both internally
+and externally.
+
+The catalog has built in support for caching when refreshing external locations
+in GitHub. This example aims to demonstrate the same behavior for `system-x`
+that we implemented earlier.
+
+```ts
+import { UrlReader } from '@backstage/backend-common';
+import { Entity, LocationSpec } from '@backstage/catalog-model';
+import {
+  results,
+  CatalogProcessor,
+  CatalogProcessorEmit,
+  CatalogProcessorCache,
+} from '@backstage/plugin-catalog-backend';
+import { CatalogProcessorParser } from '.';
+
+// Change this key to if the processor has undergone major changes
```

Comment:
Not even major changes, just adding something to the `CacheItem` and not making it optional could cause runtime failures due to bad data.